aztk/client.py

```diff
@@ -225,6 +225,32 @@ def __delete_user_on_pool(self, username, pool_id, nodes):
             futures = [exector.submit(self.__delete_user, pool_id, node.id, username) for node in nodes]
             concurrent.futures.wait(futures)
 
+    def __node_run(self, cluster_id, node_id, command, internal, container_name=None, timeout=None):
+        pool, nodes = self.__get_pool_details(cluster_id)
+        node = [node for node in nodes if node.id == node_id]
```

Comment:
would use `next(node for node in nodes if node.id == node_id)` here instead of this and the if below https://stackoverflow.com/questions/9542738/python-find-in-list